src/context/auth-context.tsx

```diff
@@ -19,55 +26,59 @@ type AuthDataProviderProps = {
 export const AuthDataProvider = ({ children }: AuthDataProviderProps) => {
     const { activeLoginid, setActiveLoginid } = useAppData();
     const { loginInfo, paramsToDelete } = URLUtils.getLoginInfoFromURL();
-    const { mutate, isSuccess } = useAuthorize();
-
-    const authorizeAccount = (loginid: string, token: string) => {
-        mutate({ authorize: token });
-        setActiveLoginid(loginid);
-        LocalStorageUtils.setValue('client.active_loginid', loginid);
-    };
-
-    const getActiveAccount = () => {
-        const accountList = LocalStorageUtils.getValue<URLUtils.LoginInfo[]>('client.account_list');
-        return accountList?.find(acc => acc.loginid === activeLoginid);
-    };
-
-    const switchAccount = (loginid: string) => {
-        if (loginid !== activeLoginid) {
-            const accountList = LocalStorageUtils.getValue<URLUtils.LoginInfo[]>('client.account_list');
-            const matchingAccount = accountList?.find(acc => acc.loginid === loginid);
-            if (matchingAccount) {
-                authorizeAccount(loginid, matchingAccount.token);
-            }
+
+    const { data, mutate, isSuccess, isPending } = useAuthorize();
+
+    const authorizeAccount = useCallback((token?: string) => {
+        if (token) mutate({ authorize: token });
+    }, []);
+
+    const switchAccount = useCallback(
+        (loginId: string) => {
+            const accountsList: AccountsList = JSON.parse(Cookies.get('accountsList') ?? '[]');
+
+            const token = accountsList.find(account => account.loginid === loginId)?.token;
+            if (!token) return;
+            authorizeAccount(token);
+
+            Cookies.set('authToken', token);
```

Comment:
@shahzaib-deriv, is everyone okay with this change ? `'client.active_loginid'` to `'authToken'` is there a thread regarding this? (I personally prefer this too actually, but...)
```suggestion
            Cookies.set('authToken', token);
```

Reply:
@jim-deriv this is how we want to go forward, we don't need to rely on storage anymore. Had a discussion with @thisyahlen-deriv and we implemented this in Tradershub v2.

Cookies are also accessible on domains without the need to syncLocalStorage script